File: java_spring-boot_spring-webflux_spring-data-jpa_lombok/src/main/java/org/example/demoapp/entity/Company.java
package org.example.demoapp.entity;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.JoinTable;
import javax.persistence.OneToMany;

import java.util.Set;

@Data
@Builder
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@NoArgsConstructor(access = AccessLevel.PRIVATE)
@Entity
public class Company {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    @OneToMany(fetch = FetchType.EAGER)
    @JoinTable(name = "company_employees",
            joinColumns = @JoinColumn(name = "company_id"),
            inverseJoinColumns = @JoinColumn(name = "employee_id"))
    private Set<Employee> employees;

    public boolean isEmployeeEmployed(long employeeId) {
        return employees.stream()
                .anyMatch(employee -> employee.getId().equals(employeeId));
    }

    public void addEmployee(Employee employee) {
        employees.add(employee);
    }

    public void removeEmployee(Employee employee) {
        employees.remove(employee);
    }

}

File: kotlin_spring-boot_spring-boot-data-jpa_spring-web_maven/src/main/kotlin/org/example/demoapp/company/Company.kt
package org.example.demoapp.company

import org.example.demoapp.employee.Employee
import javax.persistence.*

@Entity
data class Company(

        @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Long? = null,
        var name: String? = null,
        @OneToMany
        @JoinTable(name = "company_employees",
                joinColumns = [(JoinColumn(name = "company_id"))],
                inverseJoinColumns = [(JoinColumn(name = "employee_id"))])
        var employees: MutableSet<Employee> = mutableSetOf()) {

    fun alreadeEmployed(employeeId: Long): Boolean {
        return employees.any { employeeId == employeeId }
    }
}
